builder/index.html

```diff
@@ -296,12 +297,21 @@ <h3 >{{getMethodName(selectedMethod)}}
                                   <label for="os_class">Version (Minor):</label>
                                   <input type="text"  style="width:50px" ng-pattern="versionRegex" class="form-control" ng-change="updateClassContext()" ng-disabled="selectedMethod.OS_class.majorVersion === undefined || selectedMethod.OS_class.majorVersion === '' " ng-model="selectedMethod.OS_class.minorVersion"  placeholder="" name="versionMin">
                                 </div>
+                                <div ng-messages="CForm.versionForm.versionMaj.$error" class="sm-space-top" role="alert">
+                                  <div ng-message="pattern" class="text-danger">Invalid major version's number</div>
+                                </div>
+                                <div ng-messages="CForm.versionForm.versionMin.$error" role="alert">
+                                  <div ng-message="pattern" class="text-danger">Invalid minor version's number</div>
+                                </div>
                               </form>
                             </div>
                             <form name="CForm.form">
-                                <div class="form-group condition-form">
-                                  <label for="advanced">Other CFEngine classes:</label>
-                                  <input  type="text" name="cfClasses" id="advanced"  class="form-control" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" placeholder="">
+                                <div class="form-group condition-form" ng-class="{'has-error':selectedMethod.advanced_class.indexOf(' ')>0}">
+                                  <label class="control-label" for="advanced">Other CFEngine classes:</label>
+                                  <input type="text" class="form-control" name="cfClasses" id="advanced" aria-describedby="helpBlock" ng-change="updateClassContext()" ng-model="selectedMethod.advanced_class" >
```

Comment:
You should add a ng-pattern here to check the input ( value should be ng-pattern="/^\S+$/" ) and then you can use angular form validation